profile/profile.js

```diff
@@ -1,395 +1,419 @@
-				phoneNumber: $("#phoneNumberInput").val(),
-				action: "updatePersonalInformation"
+				action: 'getShifts'
 			},
 			cache: false,
 			success: function(response) {
-				if (response.success) {
-					$("#firstNameText").html($("#firstNameInput").val());
-					$("#lastNameText").html($("#lastNameInput").val());
-					$("#emailText").html($("#emailInput").val());
-					$("#phoneNumberText").html($("#phoneNumberInput").val());
-
-					$(".personal-info").find('p').show();		
-					$(".personal-info").find('input').hide();
-
-					$("#personalInformationSaveButton").hide();
-					$("#personalInformationCancelButton").hide();
-					$("#personalInformationEditButton").show();
-				} else {
-					alert(response.error);
+				for (let i = 0; i < response.shifts.length; i++) {
+					let shift = response.shifts[i];
+
+					// Create the shiftsMap
+					if (!sitesMap.has(shift.siteId)) sitesMap.set(shift.siteId, shift.title);
+					if (!shiftsMap.has(shift.siteId)) shiftsMap.set(shift.siteId, new Map());
+
+					let datesMap = shiftsMap.get(shift.siteId);
+					if (!datesMap.has(shift.dateString)) datesMap.set(shift.dateString, []);
+
+					let shiftTimes = datesMap.get(shift.dateString);
+					shiftTimes.push({
+						'shiftId': shift.shiftId,
+						'startTime': shift.startTimeString,
+						'endTime': shift.endTimeString,
+						'signedUp': shift.signedUp,
+						'userShiftId': shift.userShiftId
+					});
+
+					// Append any shifts the person is already signed up for
+					if (shift.signedUp) {
+						appendSignedUpShift(shift.title, shift.dateString, shift.startTimeString, shift.endTimeString, shift.userShiftId, shift.siteId, shift.roleName);
+					}
 				}
-
-				$("#personalInformationSaveButton").prop('disabled', false);
-				$("#personalInformationCancelButton").prop('disabled', false);
 			},
 			error: function(response) {
-				alert("Unable to communicate with the server. Please try again later.");
-				$("#personalInformationSaveButton").prop('disabled', false);
-				$("#personalInformationCancelButton").prop('disabled', false);
+				alert("Unable to load shifts. Please refresh the page in a few minutes.");
 			}
 		});
-	});
+	}
 
-	$("#addShiftButton").click(function(e) {
-		let shiftRow = $('<div></div>').addClass("add-shift-div");
-		let siteSelect = $('<select></select>').addClass("siteSelect");
-		let dateSelect = $('<select></select>').addClass("dateSelect").attr('disabled', true);
-		let timeSelect = $('<select></select>').addClass("timeSelect").attr('disabled', true);
-		let roleSelect = $('<select></select>').addClass("roleSelect");
+	function appendSignedUpShift(siteTitle, dateString, startTimeString, endTimeString, userShiftId, siteId, roleName) {
+		let shiftRow = $('<div></div>');
+		let shiftInformation = $('<span></span>').html(`${siteTitle}: ${dateString} ${startTimeString} - ${endTimeString} (${roleName})`);
+		let removeButton = $('<i></i>').addClass('fa fa-trash-o icon clickable').click(function() {
```

Comment:
I did notice that there appears to be no trash can icon in the HTML, probably because you removed font awesome. What are our choices for getting this remove button back?

Reply:
It doesn't appear that they have a trash can icon. [https://wdn.unl.edu/documentation/icons](https://wdn.unl.edu/documentation/icons)

I'll look into this further.